docker-java-core/src/main/java/com/github/dockerjava/core/DockerContextMetaFile.java

```diff
@@ -0,0 +1,82 @@
+package com.github.dockerjava.core;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class DockerContextMetaFile {
+    @JsonProperty("Name")
+    String name;
+
+    @JsonProperty("Endpoints")
+    Endpoints endpoints;
+
+    public static class Endpoints {
+        @JsonProperty("docker")
+        Docker docker;
+
+        public static class Docker {
+            @JsonProperty("Host")
+            String host;
+
+            @JsonProperty("SkipTLSVerify")
+            boolean skipTLSVerify;
+        }
+    }
+
+    public Optional<String> host() {
+        if (endpoints != null && endpoints.docker != null) {
+            return Optional.ofNullable(endpoints.docker.host);
+        }
+        return Optional.empty();
+    }
+
+    public static Optional<DockerContextMetaFile> findContextMetaFile(ObjectMapper objectMapper, File dockerConfigPath, String context) {
+        return loadAllContextMetaFiles(objectMapper, dockerConfigPath)
+            .filter(metaFile -> metaFile.endpoints != null && metaFile.endpoints.docker != null)
+            .filter(metaFile -> Objects.equals(metaFile.name, context))
+            .findFirst();
+    }
```

Comment:
in order to find the right `meta.json` once we got the context, a hash should be created using SHA-256. For example, context is `desktop-linux` then `Hashing.sha256().hashString(dockerContext, StandardCharsets.UTF_8).toString()` will return `fe9c6bd7a66301f49ca9b6a70b217107cd1284598bfc254700c989b916da791e`. So, the right path is `~/.docker/contexts/meta/fe9c6bd7a66301f49ca9b6a70b217107cd1284598bfc254700c989b916da791e/meta.json`